packages/app-lib/java/src/main/java/com/modrinth/theseus/rpc/TheseusRpc.java

```diff
@@ -0,0 +1,188 @@
+package com.modrinth.theseus.rpc;
+
+import com.google.gson.*;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.UncheckedIOException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+public final class TheseusRpc {
+    static final Gson GSON = new GsonBuilder()
+            .setStrictness(Strictness.LENIENT)
```

Comment:
Why explicitly set the parser to be lenient? `serde_json` always generates and expects strictly conforming JSON.

Reply:
The parser was explicitly set to lenient to allow reading JSON messages back-to-back without recreating the `JsonReader`, since recreating the `JsonReader` could lead to a race condition due to the buffered nature of `JsonReader`s. However, since the code has been switched to messages being in separate lines, this is no longer needed and has been set back to strict.

Reply:
Ah, nice :100:

Thanks for the explanation!